main_app/views.py

```diff
@@ -140,10 +140,37 @@ def search(request):
         return render('home')
     
 def items_details(request, item_id):
-    item = Item.objects.get(id=item_id)
+    # item = Item.objects.get(id=item_id)
+    item = get_object_or_404(Item, id=item_id)
     id_list = item.categories.all().values_list('id')
     categories_item_doesnt_have = Category.objects.exclude(id__in=id_list)
-    return render(request, 'items/detail.html', {'item':item, 'categories':categories_item_doesnt_have})
+    last_quantity_changes = QuantityLog.objects.filter(item=item).order_by('-date_log')[:5]
```

Comment:
Looks good, just some tidying up for the functionality we were speaking about with the drop down menu.